src/api/app/models/package.rb

```diff
@@ -907,14 +907,22 @@ def add_containers(opts = {})
       next unless origin_package
       origin_package.binary_releases.where(obsolete_time: nil).find_each do |binary_release|
         mc = binary_release.medium_container
-        container_list[mc] = 1 if mc
+        if mc
+          mc_update_project = mc.project.update_instance
+          # pick only one and the highest container.
+          identifier = mc_update_project.name + '/' + mc.name
```

Comment:
would you mind to break it in smaller methods?

Reply:
@adrianschroeter anything we can help you with?

Reply:
I just stared at this a bit but found no obvious way to split this. Let's move all of this out into a concern...